test/e2e/tests/test_scenarios.py

```diff
@@ -90,6 +283,61 @@ class ClusterHelper(helper.ResourceHelper):
     Overrides methods as required for custom resources.
     """
 
+    def assert_items_boto3(self, expectations: dict, state: dict) -> None:
+        if not expectations:
+            # nothing to assert as there are no expectations
+            return
+        if not state:
+            # there are expectations but no given state to validate
+            # following assert will fail and assert introspection will provide useful information for debugging
+            assert expectations == state
+
+        mdb = boto3.client("memorydb")
+        clusterName = ""
+        for (latestKey, latestValue) in state.items():
+            if latestKey == "name":
+                clusterName = latestValue
+        clusters = mdb.describe_clusters(ClusterName=clusterName)
+        cluster = clusters['Clusters'][0]
+
+        for (key, value) in expectations.items():
+            # describe_clusters API provides Name, Status, NumberOfShards, ClusterEndpoint,
+            # NodeType, EngineVersion, EnginePatchVersion, ParameterGroupName, ParameterGroupStatus,
+            # SecurityGroups, SubnetGroupName, TLSEnabled, ARN, SnapshotRetentionLimit,
+            # MaintenanceWindow, SnapshotWindow, ACLName, AutoMinorVersionUpgrade.
+            # Currently we only validate Name, NodeType, ACLName, NumShards, SnapshotWindow
+            # SnapshotRetentionLimit, MaintenanceWindow, ParameterGroupName
+            # NodeType is processed by custom logic
+            if key == "NodeType":
```

Comment:
I don't understand why we need to map the key to a boto3 response path. Why not just require that the key be the field we want to assert within the response? 
```yaml
assert:
  ACLName: <value>
```

and then for nested fields within the response, we just nest the assertion:
```yaml
assert:
  Shards:
    NumberOfNodes: <value>
```

Reply:
Creating a new class for boto3 now, so boto3 resources validation doesn't need to do together with k8s resources.